core/src/main/java/org/infinispan/interceptors/impl/NonTxIracLocalSiteInterceptor.java

```diff
@@ -189,12 +186,26 @@ private boolean skipCommand(InvocationContext ctx, FlagAffectedCommand command)
     * <p>
     * The primary owner generates a new {@link IracMetadata} and stores it in the {@link WriteCommand}.
     */
-   private void visitKey(Object key, WriteCommand command) {
+   private void visitKey(InvocationContext ctx, Object key, WriteCommand command) {
       int segment = getSegment(command, key);
       if (getOwnership(segment) != Ownership.PRIMARY) {
          return;
       }
-      IracMetadata metadata = iracVersionGenerator.generateNewMetadata(segment);
+      IracMetadata metadata = null;
+      // RemoveExpired should lose to any other conflicting write
+      if (command instanceof RemoveExpiredCommand) {
+         CacheEntry<?, ?> ce = ctx.lookupEntry(key);
+         PrivateMetadata pm = ce.getInternalMetadata();
+         if (pm != null) {
+            metadata = pm.iracMetadata();
+         }
+
+         if (metadata == null) {
+            metadata = iracVersionGenerator.generateMetadataWithCurrentVersion(segment);
```

Comment:
I'm going to take some time to think about this... not 100% convinced eh eh
also, you could check `iracVersionGenerator.getTombstone(key);` before invoking the "generate" method

Reply:
Sure, I trust you on it ;)

Reply:
I'm writing a test to make sure it is correct... but now I have a question. why do you need to send the remove to the remote site? 
`org.infinispan.xsite.irac.IracManager#checkAndTrackExpiration()` already makes sure the key is expired everywhere. You can avoid sending the remove expired to the remote site and let them expired the key locally when they access it (or by the expiration reaper?). 
Am I missing something?

Reply:
We technically don't have to, but it seemed better for us just to send a single IRAC remove expired command instead of waiting for the other site to see the value is expired then ask the original site if it is expired synchronously again.